Cogs.Publishers/Csharp/CSharpPublisher.cs

```diff
@@ -641,39 +865,61 @@ private void CreatePartialIIdentifiable(CogsModel model, string csNamespace)
         private void CreatePartialItemContainer(CogsModel model, string csNamespace)
         {
 
-            string clss = $@"using System;
+            string clss = $$"""
+using System;
 using System.Xml.Linq;
+using VDS.RDF;
 
-namespace {csNamespace}
-{{
+namespace {{csNamespace}}
+{
     /// <summary>
     /// Partial class implementation for XML generation 
     /// <summary>
     public partial class ItemContainer
-    {{ 
+    { 
         public XDocument MakeXml()
-        {{
-            XNamespace ns = ""{TargetNamespace}"";
-            XDocument xDoc = new XDocument(new XElement(ns + ""ItemContainer"", new XAttribute(XNamespace.Xmlns + "
-                + $@"""{TargetNamespacePrefix}"", ""{TargetNamespace}"")));
+        {
+            XNamespace ns = "{{TargetNamespace}}";
+            XDocument xDoc = new XDocument(new XElement(ns + "ItemContainer"));
             if (xDoc.Root != null)
-            {{
+            {
                 if (TopLevelReferences != null && TopLevelReferences.Count > 0)
-                {{
+                {
                     foreach (var item in TopLevelReferences)
-                    {{
-                        xDoc.Root.Add(new XElement(ns + ""TopLevelReference"", new XElement(ns + ""ID"", item.ID), new XElement(ns + ""TypeOfObject"", item.GetType())));
-                    }}
-                }}
+                    {
+                        xDoc.Root.Add(
+                            new XElement(ns + "TopLevelReference", 
+                                new XElement(ns + "URN", item.URN),
+                                new XElement(ns + "Agency", item.Agency),
+                                new XElement(ns + "ID", item.ID),
+                                new XElement(ns + "Version", item.Version),
+                                new XElement(ns + "TypeOfObject", item.GetType().Name)
```

Comment:
TODO after merge, get these from the identification settings instead of hard coded DDI Lifecycle entries